solr/core/src/java/org/apache/solr/core/ConfigSet.java

```diff
@@ -30,15 +32,18 @@
 
   private final SolrConfig solrconfig;
 
-  private final IndexSchema indexSchema;
+  /**Provide a Schema object on demand
+   * The first Boolean is to signify a a forcefetch
+   */
+  private final Function<Boolean, IndexSchema> indexSchema;
```

Comment:
I think using the generic interface Function here is pushing the bounds of readability/intuitiveness.  Lets just provide an interface/class instead.  Perhaps get() and getForce() with no boolean arg needed.